stat/stat.go

```diff
@@ -160,6 +160,18 @@ func CircularMean(x, weights []float64) float64 {
 	return math.Atan2(aY, aX)
 }
 
+// CircularStdDev returns the circular standard deviation of the dataset.
+//
+//	sqrt(-2 * log(sqrt((\sum_i w_i * sin(alpha_i))^2 + (\sum_i w_i * cos(alpha_i))^2) / length(x)))
+func CircularStdDev(x []float64) float64 {
+	var aX, aY float64
+	for _, v := range x {
+		aX += math.Cos(v)
+		aY += math.Sin(v)
+	}
+	return math.Sqrt(-2 * math.Log(math.Sqrt(aY*aY+aX*aX)/float64(len(x))))
```

Comment:
`math.Sqrt(-2 * math.Log(math.Hypot(aY, aX)/float64(len(x))))`

Reply:
Thank you.

Changed in a78926e7ee8285190a1b446a0fc817d2716c363c